<launch>

    <arg name="device" default = "17533279"/>
    <arg name="rate" default="20"/>

    <node name="single_reader" pkg="ptgrey_reader" type="single_reader"
         ns="pg_$(arg device)" output="screen">
        <remap from="/image_out" to="image_raw" />
        <remap from="/image_out_roi" to="image" />
        <param name="is_pub" type="bool" value="true"/>
        <param name="is_show" type="bool" value="false"/>
        <param name="is_print" type="bool" value="true"/>
        <param name="serialNum" type="int" value="$(arg device)"/>
        <param name="frameRate" type="double" value="$(arg rate)"/>
        <param name="is_auto_shutter" type="bool" value="false"/>
        <param name="shutter" type="double" value="4.5"/>
        <param name="brightness" type="double" value="3.1"/>
        <param name="exposure" type="double" value="0.8"/>
        <param name="WB_red" type="int" value="550"/>
        <param name="WB_Blue" type="int" value="810"/>
        <param name="gain" type="double" value="0.0"/>
        <rosparam>
            is_sync:   true
            is_grey:   true
            is_roi:    true
            size_x:    1280
            size_y:    1024
            center_x:  640
            center_y:  512
            cropper_x: 900
            cropper_y: 900
            down_sample_scale: 1.0

            saturation: 300
            hue:        30
            sharpness:  0
        </rosparam>
    </node>
    
    <node pkg="dji_sdk" type="dji_sdk_node" name="dji_sdk_1" output="screen">
    <!-- node parameters -->
    <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
    <param name="baud_rate" type="int" value="921600"/>
    <param name="app_id" type="int" value="1061780"/>
    <param name="app_version" type="int" value="1"/>
    <param name="align_time" type="bool" value="false"/>
    <param name="enc_key" type="string" value="b51e154455fb8fb6504a889e9955cdcee40fb8bbafc098eac32d5c260d379841"/>
    <param name="use_broadcast" type="bool" value="false"/>
    </node>



</launch>
